### PR TITLE
quick-start add create argo-server step and set default ARGO_DATAFLOW_CLUSTER_NAME env

### PR DESCRIPTION
**Pull request by marviniter**
Quick Start steps not create argo-server cause step `kubectl port-forward svc/argo-server 2746:2746` not work and there are not set ARGO_DATAFLOW_CLUSTER_NAME env cause sidecar process panic
```
panic: cluster name ("ARGO_DATAFLOW_CLUSTER_NAME") was not specifcied

goroutine 1 [running]:
main.main()
        /workspace/runner/main.go:65 +0x296
```

### CODE REVIEW

File: docs/QUICK_START.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ Deploy into the `argo-dataflow-system` namespace:
 
 ```bash
 kubectl create ns argo-dataflow-system
+kubectl apply -f https://raw.githubusercontent.com/argoproj-labs/argo-dataflow/main/config/apps/argo-server.yaml
```
Comment (alexec) [inline docs/QUICK_START.md:7]: good idea

File: manager/controllers/step_controller.go [hunk 2]
```diff
@@ -62,6 +62,10 @@ type hash struct {
 var clusterName = os.Getenv(dfv1.EnvClusterName)
 
 func init() {
+	if clusterName == "" {
```
Comment (alexec) [inline manager/controllers/step_controller.go:65]: I don't want to do this. If two clusters use the name name, there will be incorrect behaviour. It is correct for the quick-start to have "default" as the name.

**Review by alexec — CHANGES_REQUESTED** [review]

File: config/quick-start.yaml [hunk 3]
```diff
@@ -7461,6 +7461,8 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
+        - name: ARGO_DATAFLOW_CLUSTER_NAME
```
Comment (alexec) [inline config/quick-start.yaml:7464]: This file is auto generated from the quick-start directory, you’ll need to edit the source file too

Comment (marviniter) [reply config/quick-start.yaml:7464]: I had add the env config at config/manager/manager.yaml file.

**Comment by marviniter** [comment]
Hi @alexec I  had exec "make manifests" command to auto generated YAML and push the change

**Comment by alexec** [comment]
I think you still need to edit the source files.

**Review by alexec — APPROVED** [review]

**Pull request merged**